src/wallet/api/address_book.cpp

```diff
@@ -43,30 +43,43 @@ AddressBook::~AddressBook() {}
 AddressBookImpl::AddressBookImpl(WalletImpl *wallet)
     : m_wallet(wallet) {}
 
-bool AddressBookImpl::addRow(const std::string &dst_addr , const std::string &payment_id, const std::string &description)
+bool AddressBookImpl::addRow(const std::string &dst_addr , const std::string &payment_id_str, const std::string &description)
 {
-  LOG_PRINT_L2("Adding row");
-  
   clearStatus();
   
   cryptonote::account_public_address addr;
-  bool has_payment_id;
+  bool has_short_pid;
   crypto::hash8 payment_id_short;
-  if(!cryptonote::get_account_integrated_address_from_str(addr, has_payment_id, payment_id_short, m_wallet->m_wallet->testnet(), dst_addr)) {
+  if(!cryptonote::get_account_integrated_address_from_str(addr, has_short_pid, payment_id_short, m_wallet->m_wallet->testnet(), dst_addr)) {
     m_errorString = "Invalid destination address";
     m_errorCode = Invalid_Address;
     return false;
   }
 
-  crypto::hash pid32 = cryptonote::null_hash;
-  bool long_pid =  (payment_id.empty())? false : tools::wallet2::parse_long_payment_id(payment_id, pid32);
-  if(!payment_id.empty() && !long_pid) {
+  crypto::hash payment_id = cryptonote::null_hash;
+  bool has_long_pid = (payment_id_str.empty())? false : tools::wallet2::parse_long_payment_id(payment_id_str, payment_id);
+
```

Comment:
It's intentional. The short payment ID should only be used in an integrated address.